pandas/core/groupby/groupby.py

```diff
@@ -864,12 +864,27 @@ def f(g):
                 # on a string grouper column
 
                 with _group_selection_context(self):
-                    return self._python_apply_general(f)
+                    return self._python_apply_general(f, self._selected_obj)
 
         return result
 
-    def _python_apply_general(self, f):
-        keys, values, mutated = self.grouper.apply(f, self._selected_obj, self.axis)
+    def _python_apply_general(self, f: F, data: Union[Series, DataFrame]):
```

Comment:
can you add the return annotation you can use FrameOrSeriesUnion for both data and the return type